version: '3.8'
services:
  server:
    build:
      context: ./
      dockerfile: Dockerfile
    volumes:
      - ./backend:/backend
    command: uvicorn api.server:app --reload --workers 1 --host 0.0.0.0 --port 8000
    env_file:
      - .env
    tty: true
    stdin_open: true
    ports:
      - 8000:8000
    # networks:
    #   - default
    #   - solved_issues_network
    depends_on:
      - db
  db:
    image: postgres:13-alpine
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - .env
    ports:
      - 5432:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready", "-d", "solved_issues"]
      interval: 10s
      timeout: 5s
      retries: 3
  mongo:
    image: mongo:latest
    container_name: mongo
    env_file:
      - .env
    environment:
      MONGO_INITDB_ROOT_USERNAME: ${MONGO_INITDB_ROOT_USERNAME}
      MONGO_INITDB_ROOT_PASSWORD: ${MONGO_INITDB_ROOT_PASSWORD}
      MONGO_INITDB_DATABASE: ${MONGO_INITDB_DATABASE}
    volumes:
      - mongo:/data/db
    ports:
      - 6000:27017

volumes:
  postgres_data:
  mongo:
# networks:
#   solved_issues_network:
#     external:
#       name: solved_issues_network
